inf-clojure.el

```diff
@@ -725,19 +731,24 @@ The value is nil if it can't find one."
 (defun inf-clojure-show-var-documentation (var)
   "Send a form to the inferior Clojure to give documentation for VAR.
 See function `inf-clojure-var-doc-form'."
-  (interactive (inf-clojure-symprompt "Var doc" (inf-clojure-var-at-pt)))
-  (comint-proc-query (inf-clojure-proc) (format (inf-clojure-var-doc-form) var)))
+  (interactive "P")
+  (let ((var (if (inf-clojure--prompt-on-commands-p var)
+                 (car (inf-clojure-symprompt "Var doc" (inf-clojure-var-at-pt)))
+               (inf-clojure-var-at-pt))))
+    (comint-proc-query (inf-clojure-proc) (format (inf-clojure-var-doc-form) var))))
 
 (defun inf-clojure-show-var-source (var)
-  "Send a form to the inferior Clojure to give source for VAR.
-See variable `inf-clojure-var-source-form'."
-  (interactive (inf-clojure-symprompt "Var source" (inf-clojure-var-at-pt)))
-  (comint-proc-query (inf-clojure-proc) (format inf-clojure-var-source-form var)))
+  "Send a command to the inferior Clojure to give source for VAR.
+See variable `inf-clojure-var-source-command'."
+  (interactive "P")
+  (let ((var (if (inf-clojure--prompt-on-commands-p var)
```

Comment:
Read carefully the code you copied - `arg` there is the prefix arg, not the the symbol at point (like in your case). I guess what you did works because when you removed the `var` related code from `interactive` there are two bindings for `var` within the code, but this is definitely wrong. If you want to do it this way you should rename var to `prompt-for-symbl` or something.